sdk/src/main/scala/com/horizen/AbstractWallet.scala

```diff
@@ -101,6 +105,24 @@ abstract class AbstractWallet[
 
   def applyConsensusEpochInfo(epochInfo: ConsensusEpochInfo): W
 
+  def generateNextSecret[T <: Secret](secretCreator: SecretCreator[T]): Try[(W, T)] = Try {
+    require(secretCreator != null, "AbstractWallet: Secret creator must be NOT NULL.")
+    val allSecrets = this.secrets()
+    var nonce = allSecrets.size
```

Comment:
@i-Alex  secretStorage can't store initial nonce, because it stores only secrets of type SidechainTypes#SCS, type SCS = com.horizen.secret.Secret. Could you tell me please what other storage may I use instead?